diff -up ./src/pam_pkcs11/pam_pkcs11.c.no_remote_spew ./src/pam_pkcs11/pam_pkcs11.c
--- ./src/pam_pkcs11/pam_pkcs11.c.no_remote_spew	2012-02-29 15:02:45.762988673 -0800
+++ ./src/pam_pkcs11/pam_pkcs11.c	2012-02-29 15:02:50.377065817 -0800
@@ -221,7 +221,7 @@ PAM_EXTERN int pam_sm_authenticate(pam_h
 
 	  if (display && (display[0] != ':') && (display[0] != '\0'))
 	  {
-		ERR1("Remote login (from %s) is not (yet) supported", display);
+		/*ERR1("Remote login (from %s) is not (yet) supported", display); */
 		pam_syslog(pamh, LOG_ERR,
                         "Remote login (from %s) is not (yet) supported",
 			display);
